packages-exp/app-check-exp/src/internal-api.ts

```diff
@@ -292,8 +293,12 @@ function notifyTokenListeners(
           observer.next(token);
         }
       } else {
-        // Otherwise return the token, whether or not it has an error field.
-        observer.next(token);
+        // If this is a 2P listener, return the token, whether or not it
```

Comment:
Doesn't 3P observer always have an error callback because we will add one if it's not supplied. I think it will be clearer if we handle 3p and 2p separately in different blocks guarded by `observer.type`.

Reply:
I made the code a little more concise, hopefully the comments make it clear.